yarn-project/simulator/src/avm/opcodes/arithmetic.ts

```diff
@@ -59,48 +58,20 @@ export class Mul extends ThreeOperandArithmeticInstruction {
 
 export class Div extends ThreeOperandArithmeticInstruction {
   static type: string = 'DIV';
-  static readonly opcode = Opcode.DIV;
+  static readonly opcode = Opcode.DIV_8; // FIXME: needed for gas.
 
   protected compute(a: MemoryValue, b: MemoryValue): MemoryValue {
     return a.div(b);
   }
 }
 
-export class FieldDiv extends Instruction {
+// TODO: This class now temporarily has a tag, until all tags are removed.
+export class FieldDiv extends ThreeOperandArithmeticInstruction {
   static type: string = 'FDIV';
-  static readonly opcode = Opcode.FDIV;
-
-  // Informs (de)serialization. See Instruction.deserialize.
-  static readonly wireFormat: OperandType[] = [
-    OperandType.UINT8,
-    OperandType.UINT8,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-  ];
-
-  constructor(private indirect: number, private aOffset: number, private bOffset: number, private dstOffset: number) {
-    super();
-  }
-
-  public async execute(context: AvmContext): Promise<void> {
-    const memoryOperations = { reads: 2, writes: 1, indirect: this.indirect };
-    const memory = context.machineState.memory.track(this.type);
-    context.machineState.consumeGas(this.gasCost(memoryOperations));
-
-    const [aOffset, bOffset, dstOffset] = Addressing.fromWire(this.indirect).resolve(
-      [this.aOffset, this.bOffset, this.dstOffset],
-      memory,
-    );
-    memory.checkTags(TypeTag.FIELD, aOffset, bOffset);
-
-    const a = memory.getAs<Field>(aOffset);
-    const b = memory.getAs<Field>(bOffset);
+  static readonly opcode = Opcode.FDIV_8; // FIXME: needed for gas.
 
-    const dest = a.fdiv(b);
-    memory.set(dstOffset, dest);
-
-    memory.assert(memoryOperations);
-    context.machineState.incrementPc();
+  protected compute(a: Field, b: Field): Field {
+    // return (a as Field).fdiv(b as Field);
+    return a.fdiv(b);
   }
 }
```

Comment:
It's not totally clear to me why this PR let's us remove all of this code :thinking: Is it because we now have a more generic way to handle different opcode wire formats?